internal/utils/url.go

```diff
@@ -72,3 +73,19 @@ func VerifySignedURL(signedURL string, expectedPublicKey string) (bool, error) {
 
 	return true, nil
 }
+
+func SetURLSchemeIfEmpty(rawURL string) (string, error) {
```

Comment:
Calling it a setter is inaccurate because a setter should be modifying an existing resource, while this function returns a new resource.

What about calling it `GetURLWithScheme` or `GetURLStringWithScheme`?